### DISCUSSION TITLE
Canonical urls with www redirect

### DISCUSSION THREAD

**Discussion by bdlowery** [vercel/vercel#8882] [Help] [answered]
So Vercel [recommends](https://vercel.com/changelog/domains-now-include-their-www-counterpart) adding the www.example.com domain and redirecting example.com to it.

How does this effect canonical urls? Should I set the canonical url to www.example.com since everything will be redirected to the www counterpart?

**Comment by amyegan** [comment, marked as answer]
Hi @bdlowery! If you are redirecting from `example.com` to `www.example.com`, then `www.example.com` is the canonical URL and should be used for any canonical links you add to other pages. Google has some detailed documentation about duplicate content that might be helpful if you want to know more: https://developers.google.com/search/docs/crawling-indexing/consolidate-duplicate-urls
